js/ui/appDisplay.js

```diff
@@ -548,6 +548,11 @@ const AllView = new Lang.Class({
         this._addedFolderId = null;
         IconGridLayout.layout.connect('folder-added',
                                       Lang.bind(this, function(iconGridLayout, id){
+                                          // Go to last page; ideally the grid should know in
+                                          // which page the change took place and show it automatically
+                                          // which would avoid us having to navigate there directly
+                                          this.goToPage(this._grid.nPages() - 1);
```

Comment:
I am OK with this change for now, but wonder how much more difficult would it be to indicate to the icon grid layout to add the folder at the right place, instead of at the end? I guess you'd still have to take into account the case when the user right clicks e.g. outside of the grid bounds entirely, in which case you'd still need to append it.